openverse_catalog/dags/common/slack.py

```diff
@@ -213,3 +218,51 @@ def send_message(
     s = SlackMessage(username, icon_emoji, http_conn_id=http_conn_id)
     s.add_text(text, plain_text=not markdown)
     s.send(text)
+
+
+def on_failure_callback(context: dict) -> None:
+    """
+    Send an alert out regarding a failure to Slack.
+    Errors are only sent out in production and if a Slack connection is defined.
+    """
+    # Exit early if no slack connection exists
+    hook = HttpHook(http_conn_id=SLACK_CONN_ID)
+    try:
+        hook.get_conn()
+    except AirflowNotFoundException:
+        return
+
+    # Exit early if we aren't on production or if force alert is not set
+    environment = Variable.get("environment", default_var="dev")
+    force_alert = Variable.get(
+        "force_slack_alert", default_var=False, deserialize_json=True
+    )
+    if not (environment == "prod" or force_alert):
+        return
+
+    # Get relevant info
+    ti = context["task_instance"]
+    execution_date = context["execution_date"]
+    exception: Optional[Exception] = context.get("exception")
+    exception_message = ""
+
+    if exception:
+        # Forgo the alert on upstream failures
```

Comment:
nice

Reply:
If we don't do this, every task downstream of the primary failure _also_ issues an alert :joy:

Reply:
ha, I had a feeling...